specs/WebNotification.md

```diff
@@ -0,0 +1,610 @@
+  /// object, WebView will not display the default notification regardless of
+  /// the Handled property. The default value is FALSE.
+  [propput] HRESULT Handled([in] BOOL value);
+
+  /// Gets whether the `NotificationReceived` event is handled by host.
+  [propget] HRESULT Handled([out, retval] BOOL* value);
+
+  /// Returns an `ICoreWebView2Deferral` object. Use this operation to complete
+  /// the event at a later time.
+  HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** deferral);
+}
+
+/// An event handler for the `Closed` event.
+[uuid(6A0B4DE9-8CBE-4211-BAEF-AC037B1A72DE), object, pointer_default(unique)]
+interface ICoreWebView2NotificationClosedEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2Notification* sender,
+      [in] IUnknown* args);
+}
+
+/// A collection of notification actions.
+[uuid(89D8907E-18C7-458B-A970-F91F645E4C43), object, pointer_default(unique)]
+interface ICoreWebView2NotificationActionCollection : IUnknown {
+  /// The number of notification actions contained in the
+  /// ICoreWebView2NotificationActionCollection.
+  [propget] HRESULT Count([out, retval] UINT* count);
+
+  /// Gets the notification action at the given index.
+  HRESULT GetValueAtIndex([in] UINT index,
+                          [out, retval] COREWEBVIEW2_NOTIFICATION_ACTION* value);
+}
+
+/// A collection of unsigned long integers.
+[uuid(974A91F8-A309-4376-BC70-7537D686533B), object, pointer_default(unique)]
+interface ICoreWebView2UnsignedLongCollection : IUnknown {
+  /// The number of unsigned long integers contained in the
+  /// ICoreWebView2UnsignedLongCollection.
+  [propget] HRESULT Count([out, retval] UINT* count);
+
+  /// Gets the unsigned long integer at the given index.
+  HRESULT GetValueAtIndex([in] UINT index, [out, retval] UINT64* value);
+}
+
+/// This is the ICoreWebView2Notification that represents a [HTML Notification
+/// object](https://developer.mozilla.org/docs/Web/API/Notification).
+[uuid(E3F43572-2930-42EB-BD90-CAC16DE6D942), object, pointer_default(unique)]
+interface ICoreWebView2Notification : IUnknown {
+  /// Add an event handler for the `Closed` event.
+  /// This event is raised when the notification is closed by the web code.
+  HRESULT add_Closed(
+      [in] ICoreWebView2NotificationClosedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_Closed`.
+  HRESULT remove_Closed(
+      [in] EventRegistrationToken token);
+
+  /// The host may run this to report the notification has been displayed.
+  /// The NotificationReceived event is considered handled regardless of the
+  /// Handled property of the NotifificationReceivedEventArgs if the host has
+  /// run Show().
+  HRESULT Show();
+
+  /// The host may run this to report the notification has been clicked. Use
```

Comment:
`Show`, `Click`, and `Close` are designed to be called by the host so that the DOM notification events `show`, `click`, and `close` will be raised. Would `ReportShow` and such make more sense?